### PR TITLE
Improve Plugin Documentation

### LINKED ISSUE #815: Document the concept and value value range of LimitChunkCountPlugin.
Looks like there is a nice PR opportunity with low hanging fruit for someone in webpack/webpack#4178. I'm going to leave this issue here as a stub so that once we stamp down the desired behavior then we can log the defaults and a PR here can accompany a PR in the original issue

### LINKED ISSUE #1161: Basic descriptions for plugins/loaders
I really love the documentation improvements that came with the new webpack 2 website. However, I find it frustrating that some of the promoted plugins/loaders on the webpack website do not have simple descriptions of their purpose.

For example, https://github.com/webpack-contrib/worker-loader states 

> Worker loader for Webpack.

What is a Worker Loader? Why might I want it?

There is a lot of implicit knowledge that we're expecting developers to have. This makes it hard and intimidating for new users to start using the platform. My suggestion is to make sure that, at a minimum, all loaders/plugins have a short, easily understandable description of their purpose, such that a layman could make sense of their intended use.

### PR DESCRIPTION
**Pull request by skipjack**
Add lead-in descriptions (#1161) and clean up content. Most of these changes were to the local plugin docs in this repo but I also opened a few prs on external repos where necessary like [this one](https://github.com/webpack-contrib/extract-text-webpack-plugin/pull/517). Also adding documentation for #815.

Resolves #1161
Resolves #815

### CODE REVIEW

**Comment by skipjack** [comment]
@bebraw @simon04 can one of you give this a quick skim? Should be pretty much good to go.

File: content/plugins/loader-options-plugin.md [hunk 1]
```diff
@@ -1,29 +1,37 @@
 ---
 title: LoaderOptionsPlugin
 contributors:
-    - johnnyreilly
+  - johnnyreilly
+  - skipjack
 ---
 
-?> Review this content
+The `LoaderOptionsPlugin` is unlike other plugins in that it is built for migration from webpack version 1 to version 2. In webpack v2, the schema for a `webpack.config.js` became stricter; no longer open for extension by other loaders and plugins. The intention is that you pass `options` directly to loaders and plugins (i.e. `options` are __not__ global or shared.
```
Comment (bebraw) [inline content/plugins/loader-options-plugin.md]: Missing ).

File: content/plugins/loader-options-plugin.md [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
 
-However, until a loader has been updated to depend upon options being passed directly to them, the `LoaderOptionsPlugin` exists to bridge the gap.  You can configure global / shared loader options with this plugin and all loaders will receive these options.
+``` js
+new webpack.LoaderOptionsPlugin({
+  // Options...
+})
+```
 
-In the future this plugin may be removed.
+W> This plugin may be removed in the future as it only exists for migration.
```
Comment (bebraw) [inline content/plugins/loader-options-plugin.md]: I would say **will**. Stronger incentive to move.

File: content/plugins/loader-options-plugin.md [hunk 3]
```diff
@@ -1,29 +1,37 @@
...
+* `options.options.context` (`string`): The context that can be used to configure older loaders.
+* any other options allowed in a `webpack.config.js`....
+
 
-* `options.options.context` (`string`): The context that can be used to configure older loaders
-* other options as in a `webpack.config.js`....
+## Usage
 
-## Examples
+Here's a simple example of how this plugin might be used:
```
Comment (bebraw) [inline content/plugins/loader-options-plugin.md]: I would drop "simple" as that's a filler word.

**Comment by skipjack** [comment]
Merged, but again please ping me if anything here needs tweaking.

**Pull request merged**